src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -851,6 +851,11 @@ public static unsafe SocketError Bind(SafeSocketHandle handle, ProtocolType sock
             return err == Interop.Error.SUCCESS ? SocketError.Success : GetSocketErrorForErrorCode(err);
         }
 
+        public static void CancelPendingOperations(Socket socket)
+        {
+            // TODO https://github.com/dotnet/corefx/issues/24430:
```

Comment:
@wfurt, this should be the "only" thing we need to implement to get this working on Unix as well.  I didn't look into it too deeply, but it seems like we should be able to just factor out the code here:
https://github.com/dotnet/corefx/blob/7866452ae21873c16a5f8c28a83da4533fc9131c/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs#L1073-L1083
into a helper that we can use not only on close but also when CancelPendingOperations is called.  @geoffkizer would be best to comment on this, though.